string map_name
---
string result
